embedded/sql/engine.go

```diff
@@ -493,3 +495,265 @@ func normalizeParams(params map[string]interface{}) (map[string]interface{}, err
 
 	return nparams, nil
 }
+
+// CopyCatalog returns a new transaction with a copy of the current catalog.
+func (e *Engine) CopyCatalog(ctx context.Context) (*store.OngoingTx, error) {
```

Comment:
already discussed about it, it seems rare to return a new transaction as part of `CopyCatalog`, it may be better to receive it as input

Reply:
resolved